website/src/lib/chat_stream.ts

```diff
@@ -61,15 +62,22 @@ export async function* iteratorSSE(stream: ReadableStream<Uint8Array>) {
     if (!value) {
       continue;
     }
+    const full_value = unfinished_line + value;
+    const lines = full_value.split(/\r?\n/).filter(Boolean);
+    // do line buffering - otherwise leads to parsing errors
+    if (full_value[full_value.length - 1] != "\n") {
+      unfinished_line = lines.pop();
+    } else {
+      unfinished_line = "";
+    }
 
-    const fields = value
-      .split(/\r?\n/)
-      .filter(Boolean)
-      .map((line) => {
-        const colonIdx = line.indexOf(":");
-        return [line.slice(0, colonIdx), line.slice(colonIdx + 1).trimStart()];
-      });
-
-    yield Object.fromEntries(fields);
+    const fields = lines.map((line) => {
+      const colonIdx = line.indexOf(":");
+      return [line.slice(0, colonIdx), line.slice(colonIdx + 1).trimStart()];
+    });
+    // yield multiple messages distinctly
+    for (const fidx in fields) {
+      yield Object.fromEntries([fields[fidx]]);
+    }
```

Comment:
You can use `for of ` here

Reply:
> Thank you, it would be great if you could add a test case for the `iteratorSSE`.

Test cases are done, it took quite some time to fix issues with `jest` and `TextDecoderStream` and others. With my additions also others shouldn't run into problems now, as I extended the `jest.setup.js`